### ISSUE TITLE
Add `mypy` to the CI

### ISSUE THREAD

**Issue by Stannislav** [scanapi/scanapi#458]
## Feature request
`mypy` static checks were added in #456. Now `mypy` should be made part of the CI.

**Comment by Pradhvan** [comment]
@Stannislav would you want to take this next ?

**Comment by Stannislav** [comment]
> @Stannislav would you want to take this next ?

Sure, I can take that one once #456 is merged.

**Comment by Stannislav** [comment]
I just checked and the GitHub workflow in `lint.yaml` already runs `make check`, which also runs `mypy`.

So is there any other action required?

This workflow runs only on pull-requests, but not on master.

**Comment by camilamaia** [comment]
@Stannislav you are right! It is already checking. I created a new issue to enable lint on the main branch too, good idea.

https://github.com/scanapi/scanapi/issues/490
